lib/src/game/camera.dart

```diff
@@ -0,0 +1,292 @@
+/// Components marked as `isHud = true` are always rendered in screen
+/// coordinates, bypassing the camera altogether.
+///
+/// Note: right now this only applies to rendering. No transformation is
+/// done on event handling (for gestures). You have to transform it yourself
+/// using [screenToWorld].
+class Camera {
+  static const defaultCameraSpeed = 50.0; // in pixels/s
+  static const defaultShakeIntensity = 75.0; // in pixels
+  static const defaultShakeDuration = 0.3; // in seconds
+
+  /// This must be set by the Game as soon as the Camera is created.
+  ///
+  /// Do not change this reference.
+  BaseGame gameRef;
+
+  /// If set, this bypasses follow and moves the camera to a specific point
+  /// in the world.
+  ///
+  /// You can use this if you are not using follow but have a few different
+  /// camera positions or if you are using follow but you want to highlight a
+  /// spot in the world during an animation.
+  Vector2 _currentCameraDelta;
+  Vector2 _targetCameraDelta;
+
+  /// Remaining time in seconds for the camera shake.
+  double _shakeTimer = 0.0;
+
+  // Configurable parameters
+
+  double cameraSpeed = defaultCameraSpeed;
+  double shakeIntensity = defaultShakeIntensity;
+
+  /// This is the current position of the camera, ie the world coordinate that is
+  /// rendered on the top left of the screen (origin of the screen space).
+  ///
+  /// Zero means no translation is applied.
+  /// You can't change this directly; the camera will handle all ongoing
+  /// movements so they smoothly transition.
+  /// If you want to immediately snap the camera to a new place, you can do:
+  /// ```
+  ///   camera.moveTo(newPosition);
+  ///   camera.snap();
+  /// ```
+  Vector2 get position => _position.clone();
+
+  final Vector2 _position = Vector2.zero();
+
+  /// If set, the camera will "follow" this component, making sure that this
+  /// component is always rendered in a fixed position in the screen, by
+  /// immediately moving the camera to "focus" on the object.
+  ///
+  /// You probably want to set it to the player component.
+  /// Note that this is not smooth because the movement of the follow object
+  /// is assumed to be smooth.
+  PositionComponent follow;
+
+  /// Where in the screen the follow object should be.
+  ///
+  /// This is a fractional value relating to the screen size.
+  /// Changing this will smoothly move the camera to the new position
+  /// (unless you use the followObject method that immediately sets
+  /// up the camera for the new parameters).
+  Vector2 get relativeOffset => _currentRelativeOffset;
+
+  final Vector2 _currentRelativeOffset = Vector2.zero();
+  final Vector2 _targetRelativeOffset = Vector2.zero();
+
+  /// If set, this determines boundaries for the camera movement.
+  ///
+  /// The camera will never move such that a region outside the world boundaries
+  /// is shown, meaning it will stop following when the object gets close to the
```

Comment:
A wrong

Reply:
?

Reply:
There is a small typo on this comment, you wrote `an wrong` but it should be `a wrong`, right? Or am I wrong? haha